### PR TITLE
Added php-cli.ini for easy cli access.

### LINKED ISSUE #141: Loosen suhosin etc for php-cli

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #141

* added `php-cli.ini` with all the restrictions removed so that php cli can call any functions.
* create a symlink at `/usr/bin/php` so that you can just use `php` everywhere
* put a stricter `php-cli.ini` before starting the web app

### CODE REVIEW

File: php-nginx/build-scripts/build_php56.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ rm -f configure
     --enable-gd-native-ttf \
     --enable-intl=shared \
     --enable-mailparse \
-    --enable-mbstring=shared \
+    --enable-mbstring \
```
Comment (tmatsuo) [inline php-nginx/build-scripts/build_php56.sh]: There is no shared flag

File: php-nginx/build-scripts/build_php56.sh [hunk 2]
```diff
@@ -104,7 +104,7 @@ rm -f configure
     --with-xsl=shared \
     --with-fpm-user=www-data \
     --with-fpm-group=www-data \
-    --with-mysql=mysqlnd \
+    --with-mysql \
```
Comment (tmatsuo) [inline php-nginx/build-scripts/build_php56.sh]: `=mysqlnd` is not required

**Comment by tmatsuo** [comment]
The image from this PR is pushed as
gcr.io/google_appengine/php:2016-08-11_06_52

File: php-nginx/entrypoint.sh [hunk 3]
```diff
@@ -138,4 +134,7 @@ chmod -R 550 ${DOCUMENT_ROOT}
 # Change the www-data's shell back to /usr/sbin/nologin
 chsh -s /usr/sbin/nologin www-data
 
+# Remove the loose php-cli.ini
+rm ${PHP_DIR}/lib/php-cli.ini
```
Comment (bshaffer) [inline php-nginx/entrypoint.sh]: Why do we remove the `php-cli.ini` file? It would be useful for debugging purposes.

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: If you enable `exec` or the sort, it will open a door to run the loose php from web app. It can be an easy route for intruders.

Comment (bshaffer) [inline php-nginx/entrypoint.sh]: but we're locking down `exec` specifically so this doesn't happen. Until we can actually output a proper error response on the CLI, we really can't keep suhosin activated.

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: You can enable `exec` with `runtime:php`. This means it should be still as secure as possible even when you enable `exec`.

If it's for easier debugging, I can provide an alternate easy way later without loosen the security of the runtime.

Comment (bshaffer) [inline php-nginx/entrypoint.sh]: I think we can make the error message `PHP Warning: function() has been disabled for security reasons`

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: Seems like a nice idea, can you file an issue?

Comment (bshaffer) [inline php-nginx/entrypoint.sh]: We can set `display_errors=STDERR`, which will fix this issue. I'm looking into it to see if we want to set it for web as well, since STDERR should just be logged.  I verified this locally:
```php
// test.php
<?php
exec('php -r "phpinfo();"', $output, $return_var);
var_dump($output, $return_var);
// array(0) { } int(0)
```
When accessed through a web server, the CLI error message is not displayed, only if it's run on the CLI (through `stderr`)

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: Seems we should consider using that, but I want to make that change in different PR

Comment (bshaffer) [inline php-nginx/entrypoint.sh]: How about we add a `php-cli.ini` with `display_errors=STDERR`, and then I'll be happy. This means nothing changes for web configuration, or for scripts ran with `exec`, but will output helpful messages for users running scripts on the CLI.

File: php-nginx/entrypoint.sh [hunk 4]
```diff
@@ -116,18 +116,14 @@ ${PHP_DIR}/bin/php /whitelist_functions.php
 
 if [ -f "${APP_DIR}/composer.json" ]; then
     # run the composer scripts for post-deploy
-    COMPOSER_CMD="${PHP_DIR}/bin/php \
-    -d suhosin.executor.include.whitelist=phar \
-    -d suhosin.executor.func.blacklist=none \
-    -d disable_functions= \
-    -d memory_limit=-1 \
-    -d max_input_time=-1 \
+    COMPOSER_CMD="env PATH=${PATH}:${PHP_DIR}/bin ${PHP_DIR}/bin/php \
```
Comment (bshaffer) [inline php-nginx/entrypoint.sh]: this seems like the wrong place to declare our PATH var

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: Somehow other means don'e work for us. Patch welcome :)

File: php-nginx/entrypoint.sh (hunk 3 above)
Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: Done

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: PTAL

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: FYI I tested running composer with suhosin and `display_errors=STDERR`, but it didn't display anything.

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: @bshaffer 

Basically, adding `display_errors=STDERR` doesn't improve debugging experience with suhosin enabled.

I would suggest that we rollback the last commit and provide a script for better debugging in another PR. WDYT?

Comment (bshaffer) [inline php-nginx/entrypoint.sh]: @tmatsuo I tested this on the container running locally and it worked. I wonder if there's something else configured differently.

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: @bshaffer Ok, if it works, that's good. Maybe I will retry.

Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: @bshaffer Yeah, it works for me too.

**Comment by tmatsuo** [comment]
@bshaffer Anything left?

**Comment by tmatsuo** [comment]
@bshaffer OK, just added a symlink at `/usr/bin/php` for easy access when we use `su`. PTAL

File: php-nginx/entrypoint.sh [hunk 5]
```diff
@@ -116,9 +116,7 @@ ${PHP_DIR}/bin/php /whitelist_functions.php
 
 if [ -f "${APP_DIR}/composer.json" ]; then
     # run the composer scripts for post-deploy
-    COMPOSER_CMD="env PATH=${PATH}:${PHP_DIR}/bin ${PHP_DIR}/bin/php \
-    /usr/local/bin/composer \
-    --no-ansi"
+    COMPOSER_CMD="php /usr/local/bin/composer --no-ansi"
```
Comment (bshaffer) [inline php-nginx/entrypoint.sh]: nit: I think we can just call this directly w/o setting `COMPOSER_CMD` now, i.e. 

```sh
if su www-data -c php /usr/local/bin/composer --no-ansi run-script -l" \
# ...
```

**Comment by bshaffer** [comment]
one nit, otherwise lgtm!

File: php-nginx/entrypoint.sh (hunk 5 above)
Comment (tmatsuo) [inline php-nginx/entrypoint.sh]: We use ${COMPOSER_CMD} twice and I think it's fine to keep the variable.

**Comment by tmatsuo** [comment]
@bshaffer thanks!
I'm going to merge this. If you still don't like ${COMPOSER_CMD}, please open another PR.

**Comment by tmatsuo** [comment]
FYI, the new image is staged at the following locations:
gcr.io/google_appengine/php:2016-08-18_17_03
gcr.io/google_appengine/php:staging

**Comment by tmatsuo** [comment]
Also pushed to `gcr.io/google_appengine/php:latest`
